src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,94 @@
+package wolox.training.controllers;
+
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.exceptions.UsernameMismatchException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private BookRepository bookRepository;
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    @GetMapping
+    public Iterable findAll() {
+        return userRepository.findAll();
+    }
+
+    @GetMapping
+    public Optional<User> findByUsername(@PathVariable String username) {
+        return userRepository.findByUsername(username);
+    }
+
+    @PutMapping("/{username}")
+    public User updateUser(@RequestBody User user, @PathVariable long id) {
+        if (user.getId() != id) {
+            throw new UsernameMismatchException("Username does not match!");
+        }
+        try {
+            userRepository.findById(id);
+        } catch (UserNotFoundException e) {
```

Comment:
This exception should be thrown, you can remove the `try {} catch {}` and add an `.orElseThrow` in the repository method invocation :) 
We need to inform that the user was not found, so throwing the exception would be appropriate 😄

Reply:
and the same comment goes for the rest of the methods 😇